javascript/src/index.ts

```diff
@@ -813,6 +811,23 @@ export function getMissingDeps<T>(doc: Doc<T>, heads: Heads): Heads {
     return state.handle.getMissingDeps(heads)
 }
 
+export function splice<T>(doc: Doc<T>, prop: Prop, index: number, del: number, newText?: string) {
+    if (!Reflect.get(doc, IS_PROXY)) {
+      throw new RangeError("object cannot be modified outside of a change block")
+    }
+    const state = _state(doc, false)
+    const objectId = _obj(doc)
+    if (!objectId) {
+      throw new RangeError("invalid object for splice")
+    }
+    const value = `${objectId}/${prop}` //state.handle.getWithType(objectId, prop)
+    try {
+    return state.handle.splice(value, index, del, newText)
```

Comment:
A bunch of things:  first - strings are not object is javascript so I can't co-opt them with a proxy in the same way I can an Array.  So `doc.array.splice()` is possible but `doc.text.splice()` is not unless we have some kind of wrapper object like we did before.  But since all strings are text now you'd have to do `doc.name == Automerge.Text("joe")` instead of `doc.name =="joe"`.  As for users who have an old doc with a ::Str - they cannot splice these - ergo this throws an error.  Are you suggesting we have `Automerge.splice()` do a `doc.putObject( path, prop, old_string.splice(index,0,new_text))` under the hood?  That's an interesting idea ...

Reply:
I was actually thinking that we could just reimplement the logic of splice but just updating the value of the `ScalarValue::Str`. This would not merge nicely but it means the user wouldn't have to check whether the property is a string or a text.

However, the idea of doing a `putObject` is interesting, the only issue is that you might end up with a situation where two peers concurrently splice into a string and update it to a text object and you end up with a conflict. Maybe that's fine.

An alternative approach would be to provide a migration tool of some kind which automatically updates all string values to text values?

I guess my overall concern is that I'm not sure what existing users are meant to do with mixed data. If someone is writing an application which they now will receive changes containing old strings, what do they have to do?

Reply:
Doing a string.splice() will produce the exact same collision as doing a string_to_text.   But the string_to_text means the collisions will stop happening after the first sync with peers.  Lets ask PVH on this one.